### ISSUE TITLE
Fails to pass tests

### ISSUE THREAD

**Issue by copart** [RadixSeven/typescript2txt#1]
It appears that something changed in recent time that this  does not work.  I am on Fedora 23.

% make test  
ERROR: Unknown reader state (4233136) encountered in read_from
Makefile:15: recipe for target 'tests/02_passed' failed
make: **\* [tests/02_passed] Error 254

**Comment by RadixSeven** [comment]
Yep. I'm getting almost the same error with Ubuntu. (mine says that the unknown state is 4233200). I'll have to look into it. Thanks for letting me know.

**Comment by zsirmo** [comment]
`diff -Nru a/typescript2txt.cpp b/typescript2txt.cpp
--- a/typescript2txt.cpp	2011-11-17 21:34:47.000000000 +0100
+++ b/typescript2txt.cpp	2016-03-29 14:50:25.858971500 +0200
@@ -679,6 +679,7 @@
 };
 
 void Reader::read_from(std::istream& in){
+  state = SAW_NOTHING;
   while(in){
     RState next_state = SAW_NOTHING;
     int tmp_val;`

**Comment by zsirmo** [comment]
[patch](http://zsirmo.hu/work/typescript2text/uninit-state.patch)

**Comment by RadixSeven** [comment]
Sorry it is taking so long for me to get to this - even with a patch. I do intend to fix it.

**Comment by harisankarh** [comment]
This patch worked for me (Ubuntu 14.04 LTS).. Thanks @zsirmo

**Comment by ben-stephens** [comment]
Brilliant patch zsirmo. Thanks to you and RadixSeven. Works in Debian 7.3
